### PR TITLE
update to MicroProfile Context Propagation 1.3

### PR DESCRIPTION
**Pull request by Ladicek**
This includes the move from Jakarta EE 8 (javax.\*) to 9 (jakarta.\*).

### CODE REVIEW

**Comment by Ladicek** [comment]
This is very much work in progress, there's a ton of test failures that I don't understand (Weld screams about unsatisfied dependencies) and I don't understand anything about `resteasy-context-propagation`.

**Comment by manovotn** [comment]
Hmm, some of those error are really weird.

> Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.266 s <<< FAILURE! - in io.smallrye.context.test.cdi.providedBeans.ExecutorAndThreadContextInjectionTest
Error:  testInjectionWithConfigurationAndSharing(io.smallrye.context.test.cdi.providedBeans.ExecutorAndThreadContextInjectionTest)  Time elapsed: 0.265 s  <<< ERROR!
org.jboss.weld.exceptions.DeploymentException: 
WELD-001408: Unsatisfied dependencies for type ThreadContext with qualifiers @Default
  at injection point [UnbackedAnnotatedField] @Inject @ThreadContextConfig io.smallrye.context.test.FullStackResource.threadContext
  at io.smallrye.context.test.FullStackResource.threadContext(FullStackResource.java:0)

For instance this test doesn't even use the `FullStackResource` but it somehow gets included. Probably discovery?

Either way, the issue seems to be with injected TC and ME instances. E.g. functionality provided by [this CDI extension](https://github.com/smallrye/smallrye-context-propagation/blob/main/cdi/src/main/java/io/smallrye/context/inject/SmallryeContextCdiExtension.java)
@Ladicek make sure you change the `META-INF` entry for it accordingly, I don't see it in the current PR which would mean the extension is not working at all.
See https://github.com/smallrye/smallrye-context-propagation/blob/main/cdi/src/main/resources/META-INF/services/javax.enterprise.inject.spi.Extension

**Comment by Ladicek** [comment]
Ouch, I did it again! Or forgot to do it, actually. I was struggling for quite a while with MicroProfile Fault Tolerance TCK in SmallRye Fault Tolerance, only to figure out that I've changed everything _except_ the name of the `javax.enterprise.inject.spi.Extension` file in `META-INF/services`. Same here, obviously.

**Comment by Ladicek** [comment]
Fixed. Now, I really don't feel like dealing with `NoClassDefFound javax/xml/bind/JAXBException` :laughing:

**Comment by Ladicek** [comment]
Ah OK, that's RestAssured still depending on Jakarta EE 8 stuff.

**Comment by Ladicek** [comment]
Replaced the usage of RestAssured with Apache HTTP Client and the `FullStackTest` now fails with `RESTEASY003880: Unable to find contextual data of type: jakarta.ws.rs.core.UriInfo`. That _seems_ to be caused by the missing `resteasy-context-propagation`, though I didn't really investigate.

**Comment by manovotn** [comment]
@Ladicek the CI failure is related to formatting BTW.

> That seems to be caused by the missing resteasy-context-propagation, though I didn't really investigate.

Hm, don't know about that. In this codebase I only found https://github.com/smallrye/smallrye-context-propagation/blob/main/tests/src/test/java/io/smallrye/context/storage/RESTEasyContext.java
Which seems to be a fake replacement for actual context hosted in (presumably) resteasy repo. But even this one should likely suffice for tests here?

**Comment by Ladicek** [comment]
OK, so I found that RESTEasy moved a lot of MicroProfile stuff to a separate repo https://github.com/resteasy/resteasy-microprofile, including Context Propagation, and that also entails changed Maven coordinates. What used to be `org.jboss.resteasy:resteasy-context-propagation` is now `org.jboss.resteasy.microprofile:microprofile-context-propagation`. Updated. There's a test failure now which I don't really understand :-)

**Comment by Ladicek** [comment]
OK the formatter is bloody idiotic, I changed the commented out piece of code to appease its tasteless taste.

**Comment by manovotn** [comment]
> OK the formatter is bloody idiotic, I changed the commented out piece of code to appease its tasteless taste.

LOL

> FTR, there are some strange error messages that are caused by a bug in RESTEasy (https://issues.redhat.com/browse/RESTEASY-3073). I fixed that in resteasy/resteasy#3004

That's some dedication. Thanks for looking deep into this :-)

**Comment by Ladicek** [comment]
OK, all tests are passing, so I'm undrafting this. The 1st commit seems fine to me, though the 2nd one certainly deserves some discussion.

**Review by manovotn — APPROVED** [review]
Both commits look good to me. Thanks @Ladicek!

File: pom.xml [hunk 1]
```diff
@@ -93,6 +96,23 @@
                 <artifactId>rxjava</artifactId>
                 <version>${version.rxjava1}</version>
             </dependency>
+
+            <!-- SmallRye Parent 34 manages Weld to an older version that doesn't have JBoss Class File Writer 1.2.5.Final -->
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply pom.xml:100]: Parent is actually updated in this commit, and `smallrye-jakarta-parent` 34 should be fine. Except it has old Weld version, which leads to the infamous issue of https://github.com/jbossas/jboss-classfilewriter/issues/24 :-)

[the remark this answers is not in the document]
Comment (manovotn) [reply pom.xml:100]: Oh, I see. It has Weld version 4 but for some reason not the latest release :-/
Just FTR, it should be enough to just override the classfilewriter version here but having latest Weld version is likely even better.

**Comment by manovotn** [comment]
@Ladicek I just realized; shouldn't we do a version bump as part of this PR?
I can see MP CP didn't do a major version increase (which is surprising but OK) but maybe we should? Or at least minor version bump?

Cc @FroMage

**Comment by Ladicek** [comment]
Ah good point.

I'm not sure why MP ConProp went with a minor bump, all other MP specs (that I know of) have bumped a major. I'd suggest bumping a major here, too. Let me add a commit.

**Comment by Ladicek** [comment]
Ah and one other thing -- I believe what SmallRye projects do right now is having a dedicated `jakarta` branch for this effort, and `main` stays on Jakarta EE 8 for the time being. But better to coordinate with @radcortez on that front.

**Comment by Ladicek** [comment]
Bumped to `2.0.0-SNAPSHOT`.

**Review by manovotn — APPROVED** [review]
PR is LGTM, thanks for bumping the version as well.

Let's wait for @radcortez regarding the branch question before actually merging this.
I'd assume that `main` moves to `jakarta` and we create another branch for `javax` but that's apparently not what's planned :)

**Comment by xstefank** [comment]
Let me help here. So jakarta changes should go into new `jakarta` branch and there should be a major bump. - https://github.com/smallrye/smallrye-config/tree/jakarta or https://github.com/smallrye/smallrye-health/tree/jakarta

**Comment by radcortez** [comment]
> Let me help here. So jakarta changes should go into new `jakarta` branch and there should be a major bump. - https://github.com/smallrye/smallrye-config/tree/jakarta or https://github.com/smallrye/smallrye-health/tree/jakarta

+1

The reason for this is because Quarkus is still on javax and most of our development is targeting Quarkus, so I believe it would be harder for us to have to backport all the work to javax.

**Comment by manovotn** [comment]
I've created the `jakarta` branch and changed the base branch of this PR to point there.
We should be good to go now.

**Review by radcortez — APPROVED** [review]

**Pull request merged**